### PR TITLE
feat: keplr wallet

### LINKED ISSUE #1670: feat: add Keplr wallet option
### Overview

As a user, I want to use my [Keplr](https://www.keplr.app/) wallet with ShapeShift's website.

### References and additional details

- Component contents reside in `src/context/WalletProvider/Keplr`
- `src/context/WalletProvider/KeyManager.ts`

### Acceptance Criteria

- [ ] Keplr wallet is among the connect wallet options
- [ ] Uses an approved logo
- [ ] After clicking the Keplr wallet option, I will begin the wallet connect process

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Estimated effort

_No response_

### PR DESCRIPTION
**Pull request by GMSteuart**
## Description

Integrates Keplr wallet

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

Closes #1670 

## Risk

Poor UX if user is unable to use wallet as expected.

## Testing

TDB

### CODE REVIEW

**Review by pastaghost — DISMISSED** [review]

File: src/context/WalletProvider/config.ts [hunk 1]
```diff
@@ -103,4 +106,11 @@ export const SUPPORTED_WALLETS: Record<KeyManager, SupportedWalletInfo> = {
     ...DemoConfig,
     routes: [],
   },
+  [KeyManager.Keplr]: {
+    ...KeplrConfig,
+    routes: [
+      { path: '/Keplr/connect', component: KeplrConnect },
+      { path: '/Keplr/failure', component: KeplrFailure },
```
Comment (gomesalexandre) [inline src/context/WalletProvider/config.ts:113]: Inconsistent casing with the rest of the module
```suggestion
      { path: '/keplr/connect', component: KeplrConnect },
      { path: '/keplr/failure', component: KeplrFailure },
```

Routes are mapped dynamically here:
https://github.com/shapeshift/web/blob/22272912f25a792f26c788bc1548c77df9337c4e/src/context/WalletProvider/WalletViewsSwitch.tsx#L96

So it should be possible to use the same lowercase

**Review by gomesalexandre — COMMENTED** [review]
LGTM from a code perspective, one small comment. Will accept after I CR the hdwallet PR and link it here

**Review by gomesalexandre — APPROVED** [review]
LGTM, tested:

- Connecting/Disconnecting wallet works
- Switching account in Keplr disconnects the wallet
- ATOM balance in Dashboard/Asset page/Account/Accounts pages show the right amount
- Wallet/Earn/Net Worth balances in DeFi section are matching the ones in Keplr
- Opportunities in DeFi Overview/Earn are present
- Tx history is matching the one on Mintscan
- ATOM sends are working
- Delegations are working
- Undelegations are working
- Claim route isn't displayed so I haven't been able to test claims, but that is an issue in develop. Looking into it

🍬

File: src/context/WalletProvider/config.ts (hunk 1 above)
Comment (pastaghost) [reply src/context/WalletProvider/config.ts:113]: 👍

**Comment by gomesalexandre** [comment]
@pastaghost This was working locally earlier but the pair button isn't working anymore with the actual released `hdwallet-keplr`, also getting a lot of console errors

**Comment by pastaghost** [comment]
> 

Can you try again? Pairing Keplr works just fine for me in the current Fleek deployment.

**Review by gomesalexandre — APPROVED** [review]
> @gomesalexandre Can you try again? Pairing Keplr works just fine for me in the current Fleek deployment.

Jinxed cd035dd0de2a21b0acef63efa8527b94fdfe969f sending the comment, working good now, retested all the cases I've mentioned in the previous testing comment.

Noticed one non-blocking error from `useFoxyBalances.ts` that we can tackle separately after this gets in:

```shell
error TypeError: wallet.ethGetAddress is not a function
    at ChainAdapter.getAddress (EthereumChainAdapter.js:269:1)
    at useFoxyBalances.tsx:125:1
```
And a few other console errors that might not be related to this.

File: sample.env [hunk 2]
```diff
@@ -30,6 +30,7 @@ REACT_APP_FEATURE_WALLET_MIGRATION=false
 REACT_APP_FEATURE_BANXA_RAMP=false
 REACT_APP_FEATURE_FOX_PAGE=false
 REACT_APP_FEATURE_OSMOSIS=false
+REACT_APP_FEATURE_KEPLR_WALLET=false
```
Comment (0xdef1cafe) [inline sample.env:33]: added to fleek

File: src/context/WalletProvider/config.ts [hunk 3]
```diff
@@ -103,4 +106,11 @@ export const SUPPORTED_WALLETS: Record<KeyManager, SupportedWalletInfo> = {
     ...DemoConfig,
     routes: [],
   },
+  [KeyManager.Keplr]: {
```
Comment (0xdef1cafe) [inline src/context/WalletProvider/config.ts:109]: use the REACT_APP_FEATURE_KEPLR_WALLET flag here

File: src/context/WalletProvider/Keplr/components/Connect.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+import { RouteComponentProps } from 'react-router-dom'
+import { ActionTypes, WalletActions } from 'context/WalletProvider/actions'
+import { KeyManager } from 'context/WalletProvider/KeyManager'
+import { setLocalWalletTypeAndDeviceId } from 'context/WalletProvider/local-wallet'
+import { useWallet } from 'hooks/useWallet/useWallet'
+
+import { ConnectModal } from '../../components/ConnectModal'
+import { KeplrConfig } from '../config'
+
+declare global {
```
Comment (0xdef1cafe) [inline src/context/WalletProvider/Keplr/components/Connect.tsx:12]: this should probably go in a type def file outside of this component as we're extending `global` here

**Comment by 0xdef1cafe** [comment]
image

there is no user facing error if the extension is not installed

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]
please add flag, consider moving global type def to a global type def file, and handle the error gracefully if the extension is not installed

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**